omod/src/main/java/org/openmrs/module/rwandahivflowsheet/web/model/HivFlowsheetFormData.java

```diff
@@ -196,18 +196,18 @@ public Date getArvStartDate() {
     	
     	for (DrugOrder dor : rawOrders){
     		//drug start dates:
-    		Set<DrugOrder> dorListOnStartDate = ret.get(dor.getStartDate());
+    		Set<DrugOrder> dorListOnStartDate = ret.get(dor.getEffectiveStartDate());
     		if (dorListOnStartDate == null)
     			dorListOnStartDate = new LinkedHashSet<DrugOrder>();
     		//now check all orders to see if they're active on that date also, and add if so:
     		for (DrugOrder dorInner : rawOrders){
-    			if (dorInner.isCurrent(dor.getStartDate()) && !dorListOnStartDate.contains(dorInner)) //only add each DrugOrder once.
+    			if (dorInner.isActive(dor.getEffectiveStartDate()) && !dorListOnStartDate.contains(dorInner)) //only add each DrugOrder once.
     				dorListOnStartDate.add(dorInner);
     		}
     		if (dorListOnStartDate.size() > 0)
-    			ret.put(dor.getStartDate(), dorListOnStartDate);
+    			ret.put(dor.getEffectiveStartDate(), dorListOnStartDate);
     		
-    		Date endDate = dor.getDiscontinuedDate();
```

Comment:
Shouldn't we be getting the discontinuedDate the same way we do get the discontinuedReason?

Reply:
Seems #getEffectiveStopDate should handle this.

Reply:
Agreed.